test/BUILD.bazel

```diff
@@ -1,53 +1,85 @@
-    "callback.cc",
-    "static-thread-pool.cc",
-    "task.cc",
-    "closure.cc",
-    "iterate.cc",
-    "collect.cc",
-    "filter.cc",
-    "take.cc",
-    "range.cc",
-    "let.cc",
-    "flat-map.cc",
-    "concurrent.cc",
-    "do-all.cc",
-    "generator.cc",
-    "finally.cc",
-    "foreach.cc",
-    "expected.cc",
-    "unpack.cc",
-    "transformer.cc",
-    "pipe.cc",
-]
-
-eventuals_events_sources = [
-    "timer.cc",
-    "filesystem.cc",
-    "event-loop-test.h",
-    "dns-resolver.cc",
-    "signal.cc",
-]
-
-eventuals_http_sources = [
-    "http.cc",
-]
+# NOTE: we build "concurrent" tests in a separate library to
+# significantly speed up linking on platforms which prefer shared
+# libraries (e.g., macos).
+cc_library(
+    name = "concurrent-tests",
+    hdrs = [
+        "concurrent.h",
+    ],
+    # We've split all of the tests into separate files so they can be
+    # compiled in parallel which is significantly faster than having
+    # all of the tests in a single file.
+    srcs = [
+        "concurrent-success.cc",
+        "concurrent-stop.cc",
+        "concurrent-fail.cc",
+        "concurrent-fail-or-stop.cc",
+        "concurrent-stop-before-start.cc",
+        "concurrent-fail-before-start.cc",
+        "concurrent-interrupt-success.cc",
+        "concurrent-interrupt-stop.cc",
+        "concurrent-interrupt-fail.cc",
+        "concurrent-interrupt-fail-or-stop.cc",
+        "concurrent-stream-stop.cc",
+        "concurrent-stream-fail.cc",
+        "concurrent-emit-interrupt-stop.cc",
+        "concurrent-emit-interrupt-fail.cc",
+        "concurrent-emit-fail-interrupt.cc",
+        "concurrent-emit-stop-interrupt.cc",
+        "concurrent-downstream-done-both-eventuals-success.cc",
+        "concurrent-downstream-done-one-eventual-stop.cc",
+        "concurrent-downstream-done-one-eventual-fail.cc",
+        "concurrent-flat-map.cc",
+        "concurrent-moveable.cc",
+    ],
+    copts = copts(),
+    deps = [
+        "//:eventuals",
+        "@com_github_google_googletest//:gtest",
+    ],
+)
 
 cc_test(
     name = "eventuals",
-    srcs = eventuals_base_sources + eventuals_events_sources + eventuals_http_sources,
+    srcs = [
+        "event-loop-test.h",
+        "eventual.cc",
+        "stream.cc",
+        "lock.cc",
+        "repeat.cc",
+        "then.cc",
+        "conditional.cc",
+        "callback.cc",
+        "static-thread-pool.cc",
+        "task.cc",
+        "closure.cc",
+        "iterate.cc",
+        "collect.cc",
+        "filter.cc",
+        "take.cc",
+        "range.cc",
+        "let.cc",
+        "flat-map.cc",
+        "do-all.cc",
+        "generator.cc",
+        "finally.cc",
+        "foreach.cc",
+        "expected.cc",
+        "unpack.cc",
+        "transformer.cc",
+        "pipe.cc",
+        "timer.cc",
+        "filesystem.cc",
+        "dns-resolver.cc",
+        "signal.cc",
+        "http.cc",
+    ],
     copts = copts(),
     deps = [
-        "//:eventuals",
```

Comment:
I agree that this is not necessary anymore as it's getting propagated from `concurrent-tests`, but don't you think that it should still be in here just to be more explicit that we are testing eventuals library and not anything else?